lib/srv/authhandlers.go

```diff
@@ -65,6 +80,30 @@ type AuthHandlers struct {
 	Clock clockwork.Clock
 }
 
+// AuthHandlers are common authorization and authentication related handlers
+// used by the regular and forwarding server.
+type AuthHandlers struct {
+	*log.Entry
+
+	c *AuthHandlerConfig
+}
+
+// NewAuthHandlers initializes authorization and authentication handlers
+func NewAuthHandlers(config *AuthHandlerConfig) (*AuthHandlers, error) {
+	err := utils.RegisterPrometheusCollectors(prometheusCollectors...)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return &AuthHandlers{
+		c: config,
+		Entry: log.WithFields(log.Fields{
+			trace.Component:       config.Component,
+			trace.ComponentFields: log.Fields{},
```

Comment:
Feels like this can be removed? If you want to get a logger with component, this is what I usually do:

```go
log.WithField(trace.Component, config.Component)
```